pkg/splunk/test/util.go

```diff
@@ -42,3 +48,104 @@ func GetMockS3SecretKeys(name string) corev1.Secret {
 	}
 	return s3Secret
 }
+
+// MockPodExecClient to mock the PodExecClient
+type MockPodExecClient struct {
+	StdOut string
+	StdErr string
+	Err    error
+}
+
+// MockPodExecClientHandler handles the MockPodExecClient
+type MockPodExecClientHandler struct {
+	WantMockPodExecClient []*MockPodExecClient
+	GotMockPodExecClient  []*MockPodExecClient
+	MockClients           map[string]*MockPodExecClient
+}
```

Comment:
Our original idea was, to have one mock client that takes a list of commands/patterns, and serves the required return values, also, support for the validation for those commands.  Just trying to understand why we need to go with multiple clients here.